app/models/notification.rb

```diff
@@ -64,7 +64,8 @@ def self.ensure_consistency!
       DELETE
         FROM notifications n
        WHERE high_priority
-         AND notification_type NOT IN (#{types[:chat_mention].to_i}, #{types[:chat_message].to_i})
+         AND n.topic_id IS NOT NULL
+         AND n.post_number IS NOT NULL
```

Comment:
Let's limit this to just topics with no topic id. The `discourse-voting` plugin has notifications with a topic ID but no post number, and its notifications would get excluded from the consistency checker after this change. 

Happy to approve this after that change so we can get this merged.